orderer/consensus/smartbft/sync_buffer.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package smartbft
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/pkg/errors"
+)
+
+type SyncBuffer struct {
+	blockCh  chan *common.Block
+	stopCh   chan struct{}
```

Comment:
why do you need to stop the buffer? you don't have a goroutine here.

Reply:
Can you please tell me why you don't use `context` for such purposes? Like ```ctx, cancel := context.WithCancel(context.Background())```

Reply:
I would guess he wants to keep the same method signature

Reply:
> Can you please tell me why you don't use `context` for such purposes? Like `ctx, cancel := context.WithCancel(context.Background())`

It can be done with context as well, `Stop()` will simply call the `CancelFunc`.
I do want to keep the `Stop()`  function in the API.